src/main/java/org/folio/circulation/resources/PickSlipsReportResource.java

```diff
@@ -0,0 +1,153 @@
+
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+
+    final CollectionResourceClient requestsStorageClient = clients.requestsStorage();
+    final ReportRepository reportRepository = new ReportRepository(clients);
+    final ItemRepository itemRepository = new ItemRepository(clients, true, false, false);
+
+    reportRepository.getAllItemsByField(STATUS_NAME_KEY, PAGED.getValue())
+        .thenApply(r -> r.next(this::mapFetcherToItems))
+        .thenComposeAsync(r -> r.after(items -> filterRequestedItems(items, requestsStorageClient)))
+        .thenComposeAsync(r -> r.after(items -> allOf(items, itemRepository::fetchItemRelatedRecords)))
+        .thenApply(r -> r.next(items -> filterItemsByServicePoint(items, servicePointId)))
+        .thenApply(r -> r.next(this::mapResultToJson))
+        .thenApply(OkJsonResponseResult::from)
+        .thenAccept(r -> r.writeTo(routingContext.response()));
+  }
+
+  private Result<List<Item>> mapFetcherToItems(ItemsReportFetcher fetcher) {
+    return Result.succeeded(
+        fetcher.getResultListOfItems().stream()
+        .flatMap(resultListOfItem -> resultListOfItem.value().getRecords().stream())
+        .collect(Collectors.toList()));
+  }
+
+  private Result<List<Item>> filterItemsByServicePoint(List<Item> items, UUID servicePointId) {
+    return Result.succeeded(
+        items.stream()
+        .filter(item -> servicePointId.equals(item.getLocation().getPrimaryServicePointId()))
+        .collect(Collectors.toList()));
+  }
+
+  private CompletableFuture<Result<List<Result<Item>>>> filterRequestedItems(
+      List<Item> items, CollectionResourceClient client) {
+
+    List<String> itemIds = items.stream()
+        .map(Item::getItemId)
+        .collect(Collectors.toList());
+
+    List<List<String>> batches = partitionList(itemIds, UUID_BATCH_SIZE);
+
+    return allOf(batches, batch -> fetchRequestsForItems(batch, client))
+        .thenApply(r -> r.next(records -> filterRequestedItems(items, records)));
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Request>>> fetchRequestsForItems(List<String> itemIds,
+      CollectionResourceClient client) {
+
+    final Result<CqlQuery> statusQuery = exactMatch(REQUEST_STATUS_KEY, RequestStatus.OPEN_NOT_YET_FILLED.getValue());
+    final Result<CqlQuery> itemIdsQuery = exactMatchAny(REQUEST_ITEM_ID_KEY, itemIds);
+    final Result<CqlQuery> queryResult = statusQuery.combine(itemIdsQuery, CqlQuery::and);
+
+    return queryResult
+        .after(query -> client.getMany(query, itemIds.size()))
+        .thenApply(r -> r.next(this::mapResponseToRecords));
+  }
+
+  private Result<MultipleRecords<Request>> mapResponseToRecords(Response response) {
+    return MultipleRecords.from(response, Request::from, REQUESTS_KEY);
+  }
+
+  private Result<List<Result<Item>>> filterRequestedItems(List<Item> items, List<MultipleRecords<Request>> records) {
```

Comment:
Maybe something like?
```suggestion
  private Result<List<Result<Item>>> filterItemsByRequests(List<Item> items, List<MultipleRecords<Request>> records) {
```